examples/streamlit/src/run_app.py

```diff
@@ -0,0 +1,36 @@
+import os
+import subprocess
+import sys
+import webbrowser
+
+from src.config import EnvironmentalVariableNames as EnvVar, get_env
+
+
+def main():
+
+    # Getting path to python executable (full path of deployed python on Windows)
+    executable = sys.executable
+
+    # Open browser tab. May temporarily display error until streamlit server is started.
+    command = 'import time; import webbrowser; time.sleep(3); webbrowser.open("http://localhost:8501")'
+    subprocess.run(f"{executable} -c '{command}'", shell=True)
+
+    # Run streamlit server
+    path_to_main = os.path.join(
+        get_env(EnvVar.WORKING_DIR), "src", "app.py"
```

Comment:
You can use `__file__` to get the path of this file (`run_app.py`), and make the path of `app.py` from there. That way you don't need any environment variable set.

This works the same way as [using data files](https://pynsist.readthedocs.io/en/latest/faq.html#using-data-files) in the FAQ - in this case your 'data file' just happens to be code to run in streamlit.

Reply:
Got it and updated